### DISCUSSION TITLE
OpenCart and AI

### DISCUSSION THREAD

**Discussion by danielkerr** [opencart/opencart#13738] [Ideas]
How to get AI to generate product descriptions within OpenCart? or information pages?

**Comment by chulcha** [comment]
need prompt
Prompt depends on content

**Comment by felipoantonoff** [comment]
I think you need to pass the prompt, perhaps using the name of the product and what you want.

Example: Write a text of up to 1000 lines about Product A and mention its main positive points, focusing on E-commerce and greater conversion.

Something like this, this Prompt could perhaps be pre-made in the configuration or have some pre-made items that the Customer chooses when generating the description. Using wildcard/variable to identify the Product name and perhaps pass some URL or SKU that makes it easier for AI to find more data about it.

AI should help a lot, especially to generate the Meta Description based on the content, for example.

As for the information page, it should be simpler, in the configuration you can have a summary about the company/store and its work and products and, based on that, ask the AI to create an information page about the company, an information page with basic rules about return and return, terms of use page, etc. Some are easier, as the context is more general and standard.

A good way to test is by playing around with https://chat.openai.com/ and testing some Prompts to see the best results.

I also recommend, in addition to AI, analyzing the implementation of Markdown for the Information, CMS and Product pages, using something like https://github.com/erusev/parsedown as a Parser, it is just a file so it is easier to maintain , another very interesting one https://github.com/thephpleague/commonmark. I'll then consider putting it in the CMS using OCMOD if it isn't native to Opencart 4 and I think it's important to save the raw Markdown.

**Comment by yamlv** [comment]
there is a lot of examples on marketplace opencart.com how its gets done by GPT, simply download any of theese to get the idea.
